### PR TITLE
add a cat command to visitor package

### PR DESCRIPTION
**Pull request by rminnich**
This is a lot like replace. It will cat all files which match a GUID to a byte slice.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
I already know you'll make me put the data into a file. But I had to at least try.

File: pkg/visitors/cat.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/linuxboot/fiano/pkg/uefi"
+)
+
+// Cat concatenates all RAW data sections from a file into a single byte slice.
+type Cat struct {
+	// Input
+	Predicate func(f *uefi.File, name string) bool
+
+	// Output
+	Buf     []byte
```
Comment (rjoleary) [inline pkg/visitors/cat.go]: Everything looks good, except this buffer never gets printed. Here's one way to fix it:

- Change this `Buf` to an `io.Writer`
- In the test initialize it to a `bytes.Buffer`
- In the RegisterCLI, initialize it to `os.stdout`

File: pkg/visitors/cat.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// Visit applies the Extract visitor to any Firmware type.
+func (v *Cat) Visit(f uefi.Firmware) error {
+	switch f := f.(type) {
+
+	case *uefi.File:
+		return f.ApplyChildren(v)
+
+	case *uefi.Section:
+		if f.Header.Type == uefi.SectionTypeRaw {
+			v.Buf = append(v.Buf, f.Buf()[4:]...)
```
Comment (rjoleary) [inline pkg/visitors/cat.go]: Put a big TODO here about the header size being wrong.

File: pkg/visitors/cat.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return f.ApplyChildren(v)
+
+	default:
+		// Must be applied to a File to have any effect.
+		return nil
+	}
+}
+
+func init() {
+	RegisterCLI("cat", 2, func(args []string) (uefi.Visitor, error) {
```
Comment (rjoleary) [inline pkg/visitors/cat.go]: The number `2` on this line is the number of arguments. It should probably be 1.

File: pkg/visitors/cat_test.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	// Find the section and make sure it contains the expected data.
+	results := find(t, f, testGUID)
+	if len(results) != 1 {
+		t.Fatalf("got %d matches; expected 1", len(results))
+	}
+	got := results[0].Sections[0].Buf()
+	if !reflect.DeepEqual(got, catdat) {
+		t.Fatalf("want %v; got %v", catdat, got)
```
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/visitors/cat_test.go]: done haha

File: pkg/visitors/cat.go (hunk 1 above)
Comment (rminnich) [reply pkg/visitors/cat.go]: My intent here is that some program will call this and manage the Buf by itself. I guess that does not with with how the commands work ... ok, I'll change it.

File: pkg/visitors/cat.go (hunk 2 above)
Comment (rminnich) [reply pkg/visitors/cat.go]: sure

File: pkg/visitors/cat.go (hunk 3 above)
Comment (rminnich) [reply pkg/visitors/cat.go]: gotcha.

File: pkg/visitors/cat.go (hunk 1 above)
Comment (rminnich) [reply pkg/visitors/cat.go]: done

**Comment by rminnich** [comment]
Note the big test file is gone

**Pull request merged**